src/sources/udp.rs

```diff
@@ -0,0 +1,220 @@
+use crate::{
+    event::{self, Event},
+    topology::config::{DataType, GlobalOptions, SourceConfig},
+};
+use codec::{self, BytesDelimitedCodec};
```

Comment:
same here this self can be dropped as you can already reference codec without the self.